test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -9249,12 +9268,16 @@ def create_vnodes(self, name=None, attrib=None, num=1, mom=None,
                 else:
                     m_attr = None
                 self.manager(MGR_CMD_CREATE, NODE, m_attr, natvnode)
-        attrs = {}
         # only expect if vnodes were added rather than the nat vnode modified
         if expect and num > 0:
+            attrs = {}
+            n = num + len(existing_vnodes)
+            if len(existing_vnodes) > 0:
+                n = n - 1
             for k, v in attrib.items():
-                attrs[str(k) + '=' + str(self.utils.decode_value(v))] = num
-            attrs['state=free'] = num
+                attrs[str(k) + '=' + str(self.utils.decode_value(v))] = n
+            attrs['state=free'] = n
```

Comment:
This assumes all the existing vnodes are in state free.  They might be currently in use.